include/unifex/scheduler_concepts.hpp

```diff
@@ -22,40 +22,60 @@
 #include <exception>
 
 namespace unifex {
+namespace _schedule {
+  struct sender;
+  inline constexpr struct _fn {
+    template<bool>
+    struct _impl {
+      template <typename Scheduler>
+      constexpr auto operator()(Scheduler&& s) const
+          noexcept(is_nothrow_tag_invocable_v<_fn, Scheduler>)
+          -> tag_invoke_result_t<_fn, Scheduler> {
+        return tag_invoke(_fn{}, static_cast<Scheduler&&>(s));
+      }
+    };
+
+    template <typename Scheduler>
+    constexpr auto operator()(Scheduler&& s) const
+        noexcept(std::is_nothrow_invocable_v<
+            _impl<is_tag_invocable_v<_fn, Scheduler>>, Scheduler>)
+        -> std::invoke_result_t<
+            _impl<is_tag_invocable_v<_fn, Scheduler>>, Scheduler> {
+      return _impl<is_tag_invocable_v<_fn, Scheduler>>{}(
+          static_cast<Scheduler&&>(s));
+    }
 
-inline constexpr struct schedule_cpo {
-  template <typename Scheduler>
-  friend constexpr auto tag_invoke(schedule_cpo, Scheduler&& s) noexcept(
-      noexcept(static_cast<Scheduler&&>(s).schedule()))
-      -> decltype(static_cast<Scheduler&&>(s).schedule()) {
-    return static_cast<Scheduler&&>(s).schedule();
-  }
-
-  template <typename Scheduler>
-  constexpr auto operator()(Scheduler&& s) const
-      noexcept(noexcept(tag_invoke(*this, static_cast<Scheduler&&>(s))))
-          -> decltype(tag_invoke(*this, static_cast<Scheduler&&>(s))) {
-    return tag_invoke(*this, static_cast<Scheduler&&>(s));
-  }
-
-  struct schedule_sender;
-
-  constexpr schedule_sender operator()() const noexcept;
-} schedule;
+    constexpr sender operator()() const noexcept;
+  } schedule{};
 
-inline constexpr struct get_scheduler_cpo {
-  template <typename Context>
-  auto operator()(const Context &context) const noexcept
-      -> tag_invoke_result_t<get_scheduler_cpo, const Context &> {
-    static_assert(is_nothrow_tag_invocable_v<get_scheduler_cpo, const Context &>);
-    static_assert(std::is_invocable_v<
-                  decltype(schedule),
-                  tag_invoke_result_t<get_scheduler_cpo, const Context &>>);
-    return tag_invoke(*this, context);
-  }
-} get_scheduler;
+  template<>
+  struct _fn::_impl<false> {
+    template <typename Scheduler>
+    constexpr auto operator()(Scheduler&& s) noexcept(
+        noexcept(static_cast<Scheduler&&>(s).schedule()))
+        -> decltype(static_cast<Scheduler&&>(s).schedule()) {
+      return static_cast<Scheduler&&>(s).schedule();
+    }
+  };
+} // namespace _schedule
+using _schedule::schedule;
+
+namespace _get_scheduler {
+  inline constexpr struct _fn {
```

Comment:
In some places you've been naming types using a leading underscore and in other places using a trailing underscore. Is there a policy to this naming? Can we be consistent with the naming conventions?

Reply:
Yes, I was planning to clean this up. I'll switch everything to leading underscores.